app/src/main/java/org/mifos/mobile/ui/savings_account_transaction/SavingAccountsTransactionScreen.kt

```diff
@@ -0,0 +1,399 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.CompareArrows
+import androidx.compose.material3.Divider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import org.mifos.mobile.MifosSelfServiceApp
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.EmptyDataComponentWithModifiedMessageAndIcon
+import org.mifos.mobile.core.ui.component.MifosErrorComponent
+import org.mifos.mobile.core.ui.component.MifosProgressIndicatorOverlay
+import org.mifos.mobile.core.ui.theme.MifosMobileTheme
+import org.mifos.mobile.models.CheckboxStatus
+import org.mifos.mobile.models.accounts.savings.SavingsWithAssociations
+import org.mifos.mobile.models.accounts.savings.TransactionType
+import org.mifos.mobile.models.accounts.savings.Transactions
+import org.mifos.mobile.utils.CheckBoxStatusUtil
+import org.mifos.mobile.utils.CurrencyUtil
+import org.mifos.mobile.utils.DateHelper.getDateAsString
+import org.mifos.mobile.utils.Network
+import org.mifos.mobile.utils.SavingsAccountUiState
+
+@Composable
+fun SavingsAccountTransactionScreen(
+    viewModel: SavingAccountsTransactionViewModel = hiltViewModel(),
+    navigateBack: () -> Unit,
+) {
+
+    val uiState by viewModel.savingAccountsTransactionUiState.collectAsStateWithLifecycle()
+    val dialogState by viewModel.isDialogOpen.collectAsStateWithLifecycle()
+    val context: Context = LocalContext.current
+    viewModel.setCheckboxStatesList(context)
+
+    SavingsAccountTransactionScreen(
+        uiState = uiState,
+        viewModel = viewModel,
+        navigateBack = navigateBack,
+        retryConnection = { viewModel.loadSavingsWithAssociations(viewModel.savingsId) },
+        dialogState = dialogState,
+        context = context
+    )
+}
+
+@Composable
+fun SavingsAccountTransactionScreen(
+    uiState: SavingsAccountUiState,
+    viewModel: SavingAccountsTransactionViewModel,
```

Comment:
dont pass viewModel in here, interact with the viewmodel only from one function.